internal/services/mssqlmanagedinstance/mssql_managed_instance_resource.go

```diff
@@ -418,6 +417,9 @@ func (r MsSqlManagedInstanceResource) Update() sdk.ResourceFunc {
 					RequestedBackupStorageRedundancy: pointer.To(storageAccTypeToBackupStorageRedundancy(state.StorageAccountType)),
 					VCores:                           pointer.To(state.VCores),
 					ZoneRedundant:                    pointer.To(state.ZoneRedundantEnabled),
+					AdministratorLogin:               pointer.To(state.AdministratorLogin),
+					AdministratorLoginPassword:       pointer.To(state.AdministratorLoginPassword),
+					SubnetId:                         pointer.To(state.SubnetId),
```

Comment:
because subnet_id has since become mutable (idk about the admin creds), it must now be included in resource updates.